ieeeuoft/hackathon-template
# IEEE-209: Added acknowledgment form and participant onboarding

## Overview

- Resolves #IEEE-209
- Added user acceptance checking (get api thunk)
- copied over acknowledgement form from old PR (https://github.com/ieeeuoft/hackathon-template/pull/266) and converted to typescript
- added create profile thunk
- please include your feedback about the design!


## Unit Tests Created

- unit tests made for all new components and thunks


## Steps to QA

- create a normal user
- login as them into the HSS with these situations:
      - when the user hasn't completed their application, the page should show an incomplete status message
      - when the user has been waitlisted (yellow alert) or rejected (red alert), they should see appropriate messages for those review statuses
      - when the user has been accepted they should see a green message box with a "Get Started" button
            - click that button and it should bring up the acknowledgement form
            - A popup should appear when you click "Waiver of Liability and Hold Harmless Agreement" which has the contents of the waiver
            - you need to check everything on the form & input an e signature to enable the "Continue" button
            - If the user hasn't RSVP'd to the event then an error should appear
            - If the user has RSVP'd then they should see a success message with their name and team code and a link to go to their dashboard page
 - please try to break this and find other edge cases
